simtools/util/general.py

```diff
@@ -742,12 +744,11 @@ def rotate(x, y, rotation_angle_phi, rotation_angle_theta=0 * u.rad):
                 "Cannot perform coordinate transformation when x and y have different units."
             )
 
-    x_trans = np.cos(rotation_angle_theta) * (
-        x * np.cos(rotation_angle_phi) - y * np.sin(rotation_angle_phi)
-    )
-    y_trans = np.cos(rotation_angle_theta) * (
-        x * np.sin(rotation_angle_phi) + y * np.cos(rotation_angle_phi)
-    )
+    x_trans = x * np.cos(rotation_around_y_axis) * np.cos(rotation_around_z_axis) - y * np.cos(
+        rotation_around_y_axis
+    ) * np.sin(rotation_around_z_axis)
+    y_trans = x * np.sin(rotation_around_z_axis) + y * np.cos(rotation_around_z_axis)
```

Comment:
But now only the second rotation is applied only to `x`. Are you sure about it?
I really didn't check this because it is not really a standard rotation so unless we sit together it would take a bit of time for me to figure out exactly what you need. I just want to point that out so you can confirm. We can also sit together and look at it if you wish.

Reply:
yes, because the Y coordinates do not change in a rotation around the Y axis. To derive this result, I multiplied the matrix of rotation around Z and the matrix of rotation around Y:
![image](https://github.com/gammasim/gammasim-tools/assets/50831535/8a0042c4-5673-4b57-ab14-f85d3ea3da4d)

One may ask why Y axis and not X axis. To answer that one has to look to the coodinate system definition in CORSIKA:
<img width="665" alt="Screenshot 2023-06-21 at 09 14 38" src="https://github.com/gammasim/gammasim-tools/assets/50831535/0e261b0e-9f7c-4395-b0e9-213756858045">
The rotation of the plane XY in order to align it the incoming direction of the particle is around the Y axis.

Reply:
Yeah that makes sense. The confusing part is that we don't keep the z coordinate so it looks like we're "missing" an axis in the second rotation. It's obvious that at some point we will need a 3-dimensional rotation and then this function will become even more complicated... =)